fedimint-core/src/admin_client.rs

```diff
@@ -94,6 +85,64 @@ impl WsAdminClient {
             .await
     }
 
+    /// Gets the default config gen params which can be configured by the
+    /// leader, gives them a template to modify
+    pub async fn get_default_config_gen_params(&self) -> FederationResult<ConfigGenParamsRequest> {
+        self.request_auth(
+            "/get_default_config_gen_params",
+            ApiRequestErased::default(),
+        )
+        .await
+    }
+
+    /// Used by the leader to set the config gen params, after which
+    /// `ConfigGenParams` can be created
+    pub async fn set_config_gen_params(
+        &self,
+        requested: ConfigGenParamsRequest,
+    ) -> FederationResult<()> {
+        self.request_auth("/set_config_gen_params", ApiRequestErased::new(requested))
+            .await
+    }
+
+    /// Returns the consensus config gen params, followers will delegate this
+    /// call to the leader.  Once this endpoint returns successfully we can run
+    /// DKG.
+    pub async fn get_consensus_config_gen_params(
+        &self,
+    ) -> FederationResult<ConfigGenParamsConsensus> {
+        self.request(
+            "/get_consensus_config_gen_params",
+            ApiRequestErased::default(),
+        )
+        .await
+    }
+
+    /// Runs DKG, can only be called once after configs have been generated in
+    /// `get_consensus_config_gen_params`.  If DKG fails this returns a 500
+    /// error and config gen must be restarted.
+    pub async fn await_run_dkg(&self) -> FederationResult<ConfigGenParamsConsensus> {
+        self.request_auth("/await_run_dkg", ApiRequestErased::default())
+            .await
+    }
+
+    /// After DKG, returns the hash of the consensus config tweaked with our id.
+    /// We need to share this with all other peers to complete verification.
+    pub async fn get_config_verification(&self) -> FederationResult<sha256::Hash> {
+        self.request_auth("/get_config_verification", ApiRequestErased::default())
+            .await
+    }
+
+    /// After we exchange verification hashes with other peers, we call this to
+    /// confirm we all have the same consensus configs.
+    pub async fn verify_configs(
+        &self,
+        user_hashes: BTreeSet<sha256::Hash>,
```

Comment:
I don't get it - if everyone are supposed to have the same hash, shouldn't it be just one `Hash`? And if they are not - doesn't it need to know which peer id is which (in case some peer sent a duplicate or something?). If not, please add comment why it's OK / safe for the uninformed.